src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs

```diff
@@ -21,7 +21,9 @@ internal sealed class RegionDiscoveryProvider : IRegionDiscoveryProvider
         private const string RegionName = "REGION_NAME";
 
         // For information of the current api-version refer: https://docs.microsoft.com/en-us/azure/virtual-machines/windows/instance-metadata-service#versioning
-        private readonly Uri _ImdsUri = new Uri("http://169.254.169.254/metadata/instance/compute?api-version=2020-06-01");
+        private readonly string _ImdsEndpoint = "http://169.254.169.254/metadata/instance/compute";
+        private readonly string _apiVersion = "2020-06-01";
+        private Uri ImdsUri;
```

Comment:
Can we make this readonly ? Is there any point in keeping the up-to-date ImdsUri?
Also - naming is wrong.